compiler/src/formatting/format.re

```diff
@@ -25,7 +25,8 @@ type iterator_item_type =
   | IteratedValueBindings;
 type expression_parent_type =
   | GroupedExpression
-  | GenericExpression;
+  | GenericExpression
+  | NeedsLeftSideGrouped;
```

Comment:
This comment somehow went to the ether but was there when I tried to comment on this again 🤨 

This name is now pretty confusing because `NeedsLeftSideGrouped` is what checks if it needs to do something special for grouping and `GroupedExpression` doesn't. I'd maybe just call it `GroupedAccessExpression` and rename `GroupedExpression` to `GroupedInfixExpression`.